src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1021,6 +1022,13 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
                             Debug.Assert(connection is not null || !_http2Enabled);
                             if (connection is not null)
                             {
+                                if (request.IsWebSocketH2Request() && !await connection.IsConnectEnabled.Task.ConfigureAwait(false))
```

Comment:
Clever. Should this be generalized to something like `await connection.InitialSettingsReceived.Task.ConfigureAwait(false) && connection.IsConnectEnabled`?

Should there be a timeout if the settings frame isn't received?

Reply:
Even if the request times out, nothing will unblock the await on this task.

Reply:
Added timeout and `InitialSettingsReceived` task